ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/attestation/AggregatingAttestationPool.java

```diff
@@ -273,32 +275,50 @@ public synchronized List<Attestation> getAttestations(
     final Predicate<Map.Entry<UInt64, Set<Bytes>>> filterForSlot =
         (entry) -> maybeSlot.map(slot -> entry.getKey().equals(slot)).orElse(true);
 
-    final Predicate<MatchingDataAttestationGroup> filterForCommitteeIndex =
-        (group) ->
-            maybeCommitteeIndex
-                .map(index -> group.getAttestationData().getIndex().equals(index))
-                .orElse(true);
     final UInt64 slot = maybeSlot.orElse(recentChainData.getCurrentSlot().orElse(UInt64.ZERO));
     final SchemaDefinitions schemaDefinitions = spec.atSlot(slot).getSchemaDefinitions();
 
     final boolean requestRequiresAttestationWithCommitteeBits =
         schemaDefinitions.getAttestationSchema().requiresCommitteeBits();
 
+    // Committee index filter predicate is used for pre Electra attestations only (the filter is
+    // applied to the index at the attestation data level)
+    // Post Electra the index is always set to 0 at the attestation data level (the filter
+    // is rather applied to the committee bits)
+    final Predicate<MatchingDataAttestationGroup> filterForCommitteeIndex =
+        (group) ->
+            requestRequiresAttestationWithCommitteeBits
+                || maybeCommitteeIndex
+                    .map(index -> group.getAttestationData().getIndex().equals(index))
+                    .orElse(true);
+
```

Comment:
mmm.. why we need this filtering here? Can't we just filter for both pre and post electra cases in `matchingDataAttestationGroup.stream`? Maybe I'm missing something.

Reply:
We need this filtering to handle the pre Electra attestation since the filtering must done on the attestation data index in that case.
Unless you mean that we could refactor the `maybeFilterOnCommitteeIndex` in `MatchingDataAttestationGroup` (which currently filters on the committee bits) and make it filter on the attestation data index if the committee bits is empty?

Reply:
I pushed a commit where I made the maybeFilterOnCommitteeIndex handle the pre Electra attestations and hence simplifies the logic. Let me know if that's what you had in mind

Reply:
I refactored the filtering as discussed